File: app/src/main/java/com/leafchild/scopequotas/data/QuotaType.java
package com.leafchild.scopequotas.data;

/**
 * Created by: leafchild
 * Date: 03/04/2017
 * Project: ScopeQuotas
 */

public enum QuotaType {
    DAILY("Daily"),
    WEEKLY("Weekly"),
    MONTHLY("Monthly");

    QuotaType(String value) {
        this.value = value;
    }

    private String value;

    public String getValue() {
        return value;
    }

    public static QuotaType fromString(String text) {
        for (QuotaType b : QuotaType.values()) {
            if (b.value.equalsIgnoreCase(text)) {
                return b;
            }
        }
        return null;
    }

    public static QuotaType fromOrdinal(int n) {
        return values()[n];
    }
}

File: app/src/main/java/com/leafchild/scopequotas/data/QuotaType.kt
package com.leafchild.scopequotas.data

/**
 * Created by: leafchild
 * Date: 03/04/2017
 * Project: ScopeQuotas
 */

enum class QuotaType constructor(val value: String) {

    DAILY("Daily"),
    WEEKLY("Weekly"),
    MONTHLY("Monthly");

    companion object {

        fun fromString(text: String): QuotaType? {

            for (b in QuotaType.values()) {
                if (b.value.equals(text, ignoreCase = true)) {
                    return b
                }
            }
            return null
        }

        fun fromOrdinal(n: Int): QuotaType {

            return values()[n]
        }
    }
}
